### PR TITLE
Add desktop entry mod

### PR DESCRIPTION
**Pull request by A6GibKm**
I am working on a implementation of the desktop entry specification. I open this PR to start a discussion and force myself into working on it.

Eventually this should include what it is proposed in #16.

### CODE REVIEW

File: src/desktop_entry/mod.rs [hunk 1]
```diff
@@ -0,0 +1,890 @@
...
+            startup_wm_class,
+            url,
+            prefers_non_default_gpu,
+        };
+
+        desktop_entry.validate()?;
+        Ok(desktop_entry)
+    }
+
+    fn check_not_show_in(&self) -> Result<()> {
```
Comment (hxzhao527) [inline src/desktop_entry/mod.rs:203]: [Registered OnlyShowIn Environments](https://specifications.freedesktop.org/menu-spec/latest/apb.html) just list some well known, but not the all *valid*.

As [Recognized desktop entry keys](https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s06.html) say, 
> `$XDG_CURRENT_DESKTOP` should have been set by the login manager, according to the value of the DesktopNames found in the session file

So this check is not necessary.

Same to `check_only_show_in`

**Comment by whitequark** [comment]
Hi @A6GibKm! Are you still interested in pursuing this PR?

**Comment by A6GibKm** [comment]
Hi, to be honest, no not really. I don't think I have the time and I honestly forgot all about it :see_no_evil:

**Comment by whitequark** [comment]
That's alright. I will close the PR in this case and let someone else work on it if they need it.

**Pull request closed without merging**